pkg/core/hubmanager.go

```diff
@@ -110,10 +111,13 @@ func (hm *HubManager) SetHubs(hubURLs []string) {
 	// 1. create new hubs
 	// TODO handle retries and failures intelligently
 	go func() {
-		for hubURL := range hubsToCreate {
-			err := hm.create(hubURL)
-			if err != nil {
-				log.Errorf("unable to create Hub client for %s: %s", hubURL, err.Error())
+		for host := range hubsToCreate {
+			if _, ok := hm.hubs[host]; !ok {
```

Comment:
Can combine this and next line as:
 
if hub, ok := hm.hubs[host]; ok {

Reply:
We can't do this. Bcoz both are from different struct